src/page.ts

```diff
@@ -174,15 +202,25 @@ export class Page {
    * @returns An element class, allowing you to take an action upon that element
    */
   async querySelector(selector: string) {
-    const result = await this.evaluate(
-      `document.querySelector('${selector}')`,
-    );
-    if (result === null) {
-      await this.#protocol.done(
+    const result = await this.#protocol.sendWebSocketMessage<
+      Protocol.Runtime.EvaluateRequest,
+      Protocol.Runtime.EvaluateResponse
+    >("Runtime.evaluate", {
+      expression: `document.querySelector('${selector}')`,
+      includeCommandLineAPI: true,
+    });
+    if (result.result.value === null) {
+      await this.client.close(
         'The selector "' + selector + '" does not exist inside the DOM',
       );
     }
-    return new Element("document.querySelector", selector, this);
+    return new Element(
+      "document.querySelector",
```

Comment:
We should be handling Xpath above to be able to use $x too

Reply:
as we discussed earlier, for ref, the querySelector method is a rep of the browsers querySelector, so $x wont be handled here, it'll be handled in a separate method